lib/capistrano/tasks/maintenance.rake

```diff
@@ -0,0 +1,43 @@
+require 'fileutils'
+
+namespace :maintenance do
+  task :defaults do
+    set :maintenance_app_path, fetch(:app_path, "web")
+    set :maintenance_template_path, 'templates'
+    set :maintenance_basename, 'maintenance.html'
+  end
+
+  desc "Check required files and directories exist"
+  task :check do
+    on release_roles :all do
+      unless test "[ -d #{current_path}/#{fetch(:maintenance_app_path)} ]"
+          msg = "Configured Maintenance App Path : #{current_path}/#{fetch(:maintenance_app_path)} is not a directory."
+          error msg
+          fail Capistrano::FileNotFound, msg
+      end
+
+      unless test "[ -e #{fetch(:deploy_to)}/#{fetch(:maintenance_template_path)}/#{fetch(:maintenance_basename)} ]"
+          msg = "Configured Maintenance file: #{fetch(:deploy_to)}/#{fetch(:maintenance_template_path)}/#{fetch(:maintenance_basename)} is not a file."
+          error msg
+          fail Capistrano::FileNotFound, msg
+      end
+    end
+  end
+
+  desc "Enable the maintenance mode"
+  task :on do
+    on roles(:web) do
+      src = fetch(:deploy_to) + '/' + fetch(:maintenance_template_path) +'/'+ fetch(:maintenance_basename)
+      dest = release_path.join(fetch(:maintenance_app_path)).join(fetch(:maintenance_basename))
+      execute :cp, "#{src}", "#{dest}"
+    end
+  end
+
+  desc "Disable the maintenance mode"
+  task :off do
+    on roles(:web) do
+      dest = release_path.join(fetch(:maintenance_app_path)).join(fetch(:maintenance_basename))
+      execute :rm, "#{dest}"
+    end
+  end
+end
```

Comment:
Sorry for the big suggestion, but it's easier.

- Don't use `deploy_to` as hardcoded base, to let user to setup template directory properly
- Fix default to be loaded (`Load` namespace)
- Use `shared_path` as default template directory
- Do not chain join method as you can set multiple arguments
- Better to use string interpolation instead of `+` to create a path, that can have wrong result if the variable is a `path` object

```suggestion
require 'fileutils'

namespace :load do
  task :defaults do
    set :maintenance_app_path, fetch(:app_path, 'public')
    set :maintenance_template_path, -> { shared_path }
    set :maintenance_basename, 'maintenance.html'
  end
end

namespace :maintenance do
  desc "Check required files and directories exist"
  task :check do
    on roles(:web) :all do
      unless test "[ -d #{current_path}/#{fetch(:maintenance_app_path)} ]"
          msg = "Configured Maintenance App Path : #{current_path}/#{fetch(:maintenance_app_path)} is not a directory."
          error msg
          fail Capistrano::FileNotFound, msg
      end

      unless test "[ -f #{fetch(:maintenance_template_path)}/#{fetch(:maintenance_basename)} ]"
          msg = "Configured Maintenance file: #{fetch(:maintenance_template_path)}/#{fetch(:maintenance_basename)} is not a file."
          error msg
          fail Capistrano::FileNotFound, msg
      end
    end
  end

  desc "Enable the maintenance mode"
  task :on do
    on roles(:web) do
      src = "#{fetch(:maintenance_template_path)}/#{fetch(:maintenance_basename)}"
      dest = current_path.join(fetch(:maintenance_app_path), fetch(:maintenance_basename))
      execute :cp, src, dest
    end
  end

  desc "Disable the maintenance mode"
  task :off do
    on roles(:web) do
      dest = release_path.join(fetch(:maintenance_app_path), fetch(:maintenance_basename))
      execute :rm, dest
    end
  end
end
```